### PR TITLE
Updates serialization to properly convert to and from full macaroons

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7716/

**Comment by hatched** [comment]
The changes look good but should there not have been test updates as the output formats have changed?

File: jujugui/static/gui/src/app/jujulib/test-bakery.js [hunk 1]
```diff
@@ -408,6 +408,16 @@ describe('Bakery', () => {
...
+    let generateStub;
+
+    beforeEach(() => {
+      generateStub = sinon.stub(macaroonlib, 'generateMacaroons');
+      generateStub.returnsArg(0);
+    });
+
+    afterEach(() => {
+      generateStub.restore();
+    });
```
Comment (hatched) [inline jujugui/static/gui/src/app/jujulib/test-bakery.js]: \n for some breathing room :)

**Comment by jaycee** [comment]
QA:

Login should work. The profile page should load.

**Comment by jaycee** [comment]
Per a conversation in IRC, the format changes don't need further test updates b/c tests start falling over if we're inconsistent in how we serialize/deserialize.

**Comment by frankban** [comment]
The set-auth-cookie call to the charm store fails: `{"Message":"cannot unmarshal macaroons: signature has unexpected length 0","Code":""}`. So I guess we need to update how macaroons are serialized when doing that call?

**Comment by jaycee** [comment]
This is no longer needed; a simpler fix is available.

**Pull request closed without merging**